### ISSUE TITLE
[8.3.0] Use parts of HoT in 8.3.0

### ISSUE THREAD

**Issue by aarondfrancis** [handsontable/handsontable#7578]
👋 

Hey team. It used to be that I could import anything I wanted from handsontable for use in my own plugins. E.g.:

```
import Handsontable from 'handsontable';

import {isKey} from 'handsontable/commonjs/helpers/unicode';
import PluginFactory from './plugin-factory';

import {
    stopPropagation,
    stopImmediatePropagation,
    isImmediatePropagationStopped
} from 'handsontable/commonjs/helpers/dom/event';
```

But now I'm getting complaints when trying to build with 8.3.0:

```
ERROR in ./resources/js/components/worklists/sheets-navigation.js 14:0-62
Module not found: Error: Package path ./commonjs/helpers/unicode is not exported from package /pangea/node_modules/handsontable (see exports field in /pangea/node_modules/handsontable/package.json)
```

I've tried using the non `commonjs` version, since that directory is gone now. I still get the same error, just with a different path.

```
ERROR in ./resources/js/components/worklists/sheets-navigation.js 14:0-62
Module not found: Error: Package path ./../helpers/unicode is not exported from package /Users/aaron/Code/rpts2/node_modules/handsontable (see exports field in /Users/aaron/Code/rpts2/node_modules/handsontable/package.json)
```

Is there a blessed way to do this now? Or is the functionality gone?

Thanks!

**Comment by aarondfrancis** [comment]
Y'all have any ideas? I'd love to upgrade to 8.3.0

**Comment by swistach** [comment]
Hi @aarondfrancis,

Actually, in that example, you don't need to import helpers separately. You import `Handsontable` from `handsontable`, which is not a modularised version of the library.

Helpers are available under the `Handsontable.helper.*` and `Handsontable.dom.*`.

However, I suppose that at some point, you'd like to use `handsontable/base`, and then import only these modules that you need.

Currently, helpers and mixins are not exposed in "exports" as modules to import in end-applications. Moreover, we have some objections to make them public.

From our point of view, we understand these functions might help in building applications. We know that we also use them in our documentation. But, as you can notice, we use a UMD bundle that is the same as the not-modularized version. And we even do not expose documentation for helpers.

I would not say we hide them entirely in the future, but we have to solve a problem that even a small change in helper will break the Handsontable version.

I know the above might be unclear, and maybe some part of this explanation seems not to be logically consistent. Still, we would like to clean the library from the unnecessary code/functions/internal utilities and expose only this part which is crucial to work with Handsontable.

TL;DR
I'm in a camp: "If we used something in the documentation, it's public and should be exported and have its page with an example." - but it's not only my decision, and maybe it's not the best solution.

**Comment by swistach** [comment]
I close this issue for now, but you can still comment, and we appreciate your feedback.

**Comment by aarondfrancis** [comment]
Cool I'll give that a shot, thanks @swistach!
